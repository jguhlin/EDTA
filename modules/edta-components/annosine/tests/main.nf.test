nextflow_process {

    name "Test Process ANNOSINE"
    script "../main.nf"
    process "ANNOSINE"

    tag "modules"
    tag "modules_nfcore"
    tag "annosine"

    test("actinidia_chinensis-genome_21_fasta_gz") {

        config './nextflow.config'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['actinidia_chinensis']['genome']['genome_21_fasta_gz'], checkIfExists: true)
                ]
                input[1] = 3
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("actinidia_chinensis-genome_21_fasta_gz-stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['actinidia_chinensis']['genome']['genome_21_fasta_gz'], checkIfExists: true)
                ]
                input[1] = 3
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}