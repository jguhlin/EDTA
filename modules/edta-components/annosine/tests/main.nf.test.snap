{
    "actinidia_chinensis-genome_21_fasta_gz-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e4a37dd3eccd5ff39c2262542db40e98"
                ],
                "fa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e4a37dd3eccd5ff39c2262542db40e98"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-03T16:58:04.62843"
    },
    "actinidia_chinensis-genome_21_fasta_gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fa:md5,fc3bd39555133a1b50623bccc7d4cf9b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e4a37dd3eccd5ff39c2262542db40e98"
                ],
                "fa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fa:md5,fc3bd39555133a1b50623bccc7d4cf9b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e4a37dd3eccd5ff39c2262542db40e98"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-03T16:57:59.09361"
    }
}